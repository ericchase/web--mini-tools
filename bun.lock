{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "devDependencies": {
        "@types/bun": "1.2.20",
        "@types/chrome": "0.1.4",
        "@types/node": "24.3.0",
        "tree-kill": "1.2.2",
      },
    },
  },
  "packages": {
    "@types/bun": ["@types/bun@1.2.20", "", { "dependencies": { "bun-types": "1.2.20" } }, "sha512-dX3RGzQ8+KgmMw7CsW4xT5ITBSCrSbfHc36SNT31EOUg/LA9JWq0VDdEXDRSe1InVWpd2yLUM1FUF/kEOyTzYA=="],

    "@types/chrome": ["@types/chrome@0.1.4", "", { "dependencies": { "@types/filesystem": "*", "@types/har-format": "*" } }, "sha512-vfISO7SPppN3OKVUqWujtZ4vux3nhDqKaHYEHgfQuPARHuWJ3jjyc1s13H0ckzEc86/neTkCl1TeW72UK6jYKA=="],

    "@types/filesystem": ["@types/filesystem@0.0.36", "", { "dependencies": { "@types/filewriter": "*" } }, "sha512-vPDXOZuannb9FZdxgHnqSwAG/jvdGM8Wq+6N4D/d80z+D4HWH+bItqsZaVRQykAn6WEVeEkLm2oQigyHtgb0RA=="],

    "@types/filewriter": ["@types/filewriter@0.0.33", "", {}, "sha512-xFU8ZXTw4gd358lb2jw25nxY9QAgqn2+bKKjKOYfNCzN4DKCFetK7sPtrlpg66Ywe3vWY9FNxprZawAh9wfJ3g=="],

    "@types/har-format": ["@types/har-format@1.2.16", "", {}, "sha512-fluxdy7ryD3MV6h8pTfTYpy/xQzCFC7m89nOH9y94cNqJ1mDIDPut7MnRHI3F6qRmh/cT2fUjG1MLdCNb4hE9A=="],

    "@types/node": ["@types/node@24.3.0", "", { "dependencies": { "undici-types": "~7.10.0" } }, "sha512-aPTXCrfwnDLj4VvXrm+UUCQjNEvJgNA8s5F1cvwQU+3KNltTOkBm1j30uNLyqqPNe7gE3KFzImYoZEfLhp4Yow=="],

    "@types/react": ["@types/react@19.1.10", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-EhBeSYX0Y6ye8pNebpKrwFJq7BoQ8J5SO6NlvNwwHjSj6adXJViPQrKlsyPw7hLBLvckEMO1yxeGdR82YBBlDg=="],

    "bun-types": ["bun-types@1.2.20", "", { "dependencies": { "@types/node": "*" }, "peerDependencies": { "@types/react": "^19" } }, "sha512-pxTnQYOrKvdOwyiyd/7sMt9yFOenN004Y6O4lCcCUoKVej48FS5cvTw9geRaEcB9TsDZaJKAxPTVvi8tFsVuXA=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "tree-kill": ["tree-kill@1.2.2", "", { "bin": { "tree-kill": "cli.js" } }, "sha512-L0Orpi8qGpRG//Nd+H90vFB+3iHnue1zSSGmNOOCh1GLJ7rUKVwV2HvijphGQS2UmhUZewS9VgvxYIdgr+fG1A=="],

    "undici-types": ["undici-types@7.10.0", "", {}, "sha512-t5Fy/nfn+14LuOc2KNYg75vZqClpAiqscVvMygNnlsHBFpSXdJaYtXMcdNLpl/Qvc3P2cB3s6lOV51nqsFq4ag=="],
  }
}
