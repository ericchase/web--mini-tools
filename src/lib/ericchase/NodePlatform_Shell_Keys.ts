import { Core_Array_Split } from './Core_Array_Split.js';
import { Core_JSON_Parse_Raw_String } from './Core_JSON_Parse_Raw_String.js';
import { Core_String_Split } from './Core_String_Split.js';
import { Core_String_Split_Multiple_Spaces } from './Core_String_Split_Multiple_Spaces.js';

/**
 * The Core_JSON_Parse_Raw_String(String.raw``)s are to keep bundlers (i.e Bun)
 * from replacing the unicode code points with an alternative representation.
 */
const SHELL__GENERALASCIICODES = Create_Ascii_Code_Map(String.raw`
| BEL | \u0007 | Terminal bell
| BS  | \u0008 | Backspace
| HT  | \u0009 | Horizontal TAB
| LF  | \u000A | Linefeed (newline)
| VT  | \u000B | Vertical TAB
| FF  | \u000C | Formfeed (also: New page NP)
| CR  | \u000D | Carriage return
| ESC | \u001B | Escape character
| DEL | \u007F | Delete character
`);

const SHELL__KEYS_ESC = SHELL__GENERALASCIICODES.ESC;

const SHELL__KEYS = {
  ARROWS: {
    DOWN: Core_JSON_Parse_Raw_String(String.raw`\u001B[B`),
    LEFT: Core_JSON_Parse_Raw_String(String.raw`\u001B[D`),
    RIGHT: Core_JSON_Parse_Raw_String(String.raw`\u001B[C`),
    UP: Core_JSON_Parse_Raw_String(String.raw`\u001B[A`),
  },
  GENERAL: {
    BEL: SHELL__GENERALASCIICODES.BEL,
    BS: SHELL__GENERALASCIICODES.BS,
    CR: SHELL__GENERALASCIICODES.CR,
    CSI: `${SHELL__KEYS_ESC}[`,
    DCS: `${SHELL__KEYS_ESC}P`,
    DEL: SHELL__GENERALASCIICODES.DEL,
    ESC: SHELL__KEYS_ESC,
    FF: SHELL__GENERALASCIICODES.FF,
    HT: SHELL__GENERALASCIICODES.HT,
    LF: SHELL__GENERALASCIICODES.LF,
    OSC: `${SHELL__KEYS_ESC}]`,
    VT: SHELL__GENERALASCIICODES.VT,
  },
  SIGINT: Core_JSON_Parse_Raw_String(String.raw`\u0003`), // Kill the currently running task in terminal.
};

function Create_Ascii_Code_Map(table: string): Record<string, string> {
  const map: Record<string, string> = {};
  for (const [name, code] of Core_Array_Split(Core_String_Split(table.trim(), '|', true), 3)) {
    map[name.trim()] = Core_JSON_Parse_Raw_String(Core_String_Split_Multiple_Spaces(code, true)[0]);
  }
  return map;
}

export const NodePlatform_Shell_Keys = SHELL__KEYS;
