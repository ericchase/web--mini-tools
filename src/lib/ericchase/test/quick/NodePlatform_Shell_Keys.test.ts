import { expect, test } from 'bun:test';
import * as module from '../../NodePlatform_Shell_Keys.js';

/**
 * To create/update snapshots, run `bun test --update-snapshots`.
 */
test('NodePlatform_Shell_Keys', () => {
  expect(JSON.stringify(module['NodePlatform_Shell_Keys'])).toMatchSnapshot();
});
