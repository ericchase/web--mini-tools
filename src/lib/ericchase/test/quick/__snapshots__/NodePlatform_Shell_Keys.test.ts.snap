// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePlatform_Shell_Keys 1`] = `"{"ARROWS":{"DOWN":"\\u001b[B","LEFT":"\\u001b[D","RIGHT":"\\u001b[C","UP":"\\u001b[A"},"GENERAL":{"BEL":"\\u0007","BS":"\\b","CR":"\\r","CSI":"\\u001b[","DCS":"\\u001bP","DEL":"","ESC":"\\u001b","FF":"\\f","HT":"\\t","LF":"\\n","OSC":"\\u001b]","VT":"\\u000b"},"SIGINT":"\\u0003"}"`;
